### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by malt-r** [Programmiermethoden/Dungeon#29] [Announcements]
Folgende Meilensteine sind bereits im Projektantrag definiert, die Meilensteine sollten allerdings noch verfeinert werden und wir sollten überprüfen, ob die Reihenfolge der jeweiligen Tätigkeiten sinnvoll ist.

# Phase 1: Vorbereitung

**Meilenstein 1 (30.11.2022)**: Referenzaufgaben und ein konkretes Bewertungsschema
liegen vor, das Systemdesign ist definiert und die DSL ist festgelegt.
 
**Meilenstein 2 (31.03.2023)**: Das Basissystem ist einsatzfähig, man kann in Spielen
Bewertungspunkte sammeln und die Level/Aufgaben können als Petri-Netze
visualisiert werden.

**Meilenstein 3 (15.05.2023)**: Das Interpreter ist einsatzfähig, Referenzaufgaben
können mit der DSL programmiert werden.

**Meilenstein 4 (31.05.2023)**: Es sind Aufgaben für zwei verschiedene
Lehrveranstaltungen im Framework umgesetzt und als Spiel für die
Studierenden verfügbar.

# Phase 2: Einsatz & Erweiterung

## Einsatz & Evaluierung in Lehrveranstaltungen 

**Meilenstein 5 (31.08.2023)**: Das Framework wurde in zwei Lehrveranstaltungen
eingesetzt und jeweils durch die Studierenden evaluiert.

## Erweiterung

**Meilenstein 6 (30.06.2023)**: Studierende können über das High-Score-System in
einen Wettstreit treten.

**Meilenstein 7 (31.08.2023)**: Studierende können ihre Lösungen für Code-Aufgaben
über das CI/CD-System abgeben und ein automatisiertes Feedback erhalten.
Die Anbindung an das High-Score-System ist etabliert.

**Comment by AMatutat** [comment]
Ich würde zu Meilenstein 1 noch ergänzen, dass das Spiel mit der Mehrheit der Spiel-Features umgesetzt sein soll.
